pkg/serverless/trace/span_modifer_test.go

```diff
@@ -91,3 +91,39 @@ func TestInferredSpanFunctionTagFiltering(t *testing.T) {
 	_, tagOriginSelfSpanHasGlobalTags := tp.Chunks[0].Spans[0].GetMeta()["function_arn"]
 	assert.False(t, tagOriginSelfSpanHasGlobalTags, "A span with meta._inferred_span.tag_origin = self should not get global tags")
 }
+
+func TestLambdaSpanChan(t *testing.T) {
```

Comment:
This test fails if the name of the span changes, which you can confirm, as I did:
<img width="1210" alt="image" src="https://user-images.githubusercontent.com/1598537/233179931-10cb92fa-f450-445f-a940-42c8a846abfd.png">
by changing:
`tp.Chunks[0].Spans[0].Name = "aws.lambdaFoo"`

The issue in the original commit is that the channel filter conditional was too broad, so I would expect it to pass with the reverted code.

Reply:
Alternatively I could modify the test with a similar case and assert that no span is passed down the `lambdaSpanChan`, but I fear that the span modifier could be changed again and the negative-case test would still pass.

If you'd like, I'm happy to add both, which should cover this exhaustively.

Reply:
This is quite complex, and I'll admit I do not completely comprehend exactly what is going on here.

Personally, I really like test driven development. When I find a bug, I create a failing test, then fix the bug. If later on down the road the code changes in such a way that the original test could never fail, well, then didn't we still accomplish our goal?

I'll leave it up to you and what you think is best, especially since you'll be the primary one supporting this feature going forward. I would just like to see a test that would fail if we for some reason create a span that has cyclical hierarchy.

Reply:
I'll add an additional case, which writes 3 random spans, with only 1 target span, and assert we receive only 1 span.